src/node/blockstorage.cpp

```diff
@@ -933,7 +933,7 @@ bool BlockManager::WriteBlockUndo(const CBlockUndo& blockundo, BlockValidationSt
     // Write undo information to disk
     if (block.GetUndoPos().IsNull()) {
         FlatFilePos pos;
-        const unsigned int blockundo_size{static_cast<unsigned int>(GetSerializeSize(blockundo))};
+        const auto blockundo_size{static_cast<uint32_t>(GetSerializeSize(blockundo))};
```

Comment:
In commit "refactor: clear up blockstorage/streams in preparation for optimization" (056cb3c0d2efb86447b7ff7788c206e2e7d72c12)

Introducing `uint32_t` in this commit seems like a regression to me. Maybe I am missing something but it seems like previously code was only using `size_t` and `unsigned` types, and now it is still using both of these types and `uint32_t` is being added as a third type for no clear reason. (GetSerializeSize still returns size_t and FlatFilePos and related functions still use `unsigned`) Would suggest just keeping previous types since it seemed better, and in long run just switching to `size_t` everywhere.

Or feel free it ignore this comment if I am missing something, since this is not important.

Reply:
This was changed as a consequence of adjusting the constants to avoid a few casts in https://github.com/bitcoin/bitcoin/pull/31551#discussion_r2036785539
But I'll be the first to admit that I'm not exactly sure why we're using both `unsigned int` and `uint32_t` (the latter seems more precise to me, but I know you prefer the more general ones).
If it's not a big deal, I'd like to tackle this in a follow-up PR, if needed, we've been editing and ACK-ig this PR for quite a few rounds now.